src/avatar_action.cpp

```diff
@@ -815,6 +828,16 @@ void avatar_action::plthrow( avatar &you, int pos,
         add_msg( m_info, _( "You can't effectively throw while you're in your shell." ) );
         return;
     }
+    if( you.has_effect( effect_riding ) && g->u.mounted_creature ) {
+        auto mons = g->u.mounted_creature.get();
+        if( mons->has_flag( MF_RIDEABLE_MECH ) ) {
+            if( !mons->check_mech_powered() ) {
```

Comment:
Similar here: a check for
- effect and mounted creature and
- ( creature has mech flag and has power ) or (does not have mech flag)

inside a separate functions would be nice.

That would also ensure that all places use the same logic to check this kind of thing.

Reply:
It would require two checks anyway ? 
Check if mech, and then check if has power.
I suppose one function could return an enum for three possibilities? 
1. is not mech, 2. is mech but has no power, or 3. is mech and has power ?